kernel/src/main/scala/me/rexim/morganey/syntax/LambdaParser.scala

```diff
@@ -27,38 +28,65 @@ class LambdaParser extends JavaTokenParsers with ImplicitConversions {
   def variable: Parser[LambdaVar] =
     identifier.r ^^ { LambdaVar }
 
-  def numericLiteral: Parser[LambdaTerm] =
+  def validNumberLiteral: Parser[Int] =
     numberLiteral.r ^? ({
-      case IntMatcher(x) => ChurchNumberConverter.encodeNumber(x)
+      case IntMatcher(x) => x
     }, { (rawInt) => s"`$rawInt' is too big"})
 
-  def characterLiteral: Parser[LambdaTerm] = (
+  def numericLiteral: Parser[LambdaTerm] =
+    validNumberLiteral ^^ ChurchNumberConverter.encodeNumber
+
+  def validCharacterLiteral: Parser[Char] = (
     escapedCharLiteral.r ^^ { s =>
-      ChurchNumberConverter.encodeNumber(escapeSequences(s charAt 2))
-    }
-    | symbolCharLiteral.r ^^ { s =>
-      ChurchNumberConverter.encodeNumber(s charAt 1)
+      escapeSequences(s charAt 2)
     }
+    | symbolCharLiteral.r ^^ { _ charAt 1 }
   )
 
+  def characterLiteral: Parser[LambdaTerm] =
+    validCharacterLiteral ^^ { c => ChurchNumberConverter.encodeNumber(c.toInt) }
+
   def stringLiteralTerm: Parser[LambdaTerm] =
     stringLiteral ^^ { s =>
       ChurchPairConverter.encodeString(unquoteString(s))
     }
 
+  private def range[T](p: => Parser[T])(implicit ev: Integral[T]): Parser[LambdaTerm] = {
+    val q = p
+    val parser = q ~ opt(comma ~> q) ~ (rangeOperator ~> q)
+    parser ^^ { case start ~ next ~ exit =>
+      val step  = next.map(ev.minus(_, start)).getOrElse(ev.one)
+      val range = NumericRange.inclusive(start, exit, step).toList
+      val nums  = range map ev.toInt map ChurchNumberConverter.encodeNumber
+      ChurchPairConverter.encodeList(nums)
+    }
+  }
```

Comment:
> I think the fact that we need morganey.macro._ so often in the places where it's not accessible only hints us that we are doing something wrong. Maybe we have to move everything that is in the Church Converters to morganey.macro._ not sure yet. Let me create an issue that.

I think the best solution for that is be to move `Liftable` and `Unliftable` from `macros` to `kernel`.

The implementation of `ChurchNumberConverter` and `ChurchPairConverter` could be moved into the definition of the typeclasses. That would allow us to deprecate `ChurchNumberConverter` and `ChurchPairConverter` and use the typeclasses instead. That would define a much cleaner interface for encoding (lifting) and decoding (unlifting) lambda terms.